cmake/USDPluginTools.cmake

```diff
@@ -295,11 +304,24 @@ function(usd_python_test TEST_TARGET PYTHON_FILE)
 
     # Set-up runtime environment variables for the test.
     # The paths refer to the build tree (which mirrors the final installation).
+    set(TEST_PYTHON_PATH 
+        "PYTHONPATH=${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/python" 
+        "${USD_ROOT}/${CMAKE_INSTALL_LIBDIR}/python" 
+        "$ENV{PYTHONPATH}"
+        "\\;PXR_PLUGINPATH_NAME=${_TEST_PXR_PLUGIN_PATH}")  # For Linux we need the semicolon in front
+    
+    if (MSVC)
+        list(JOIN TEST_PYTHON_PATH "\\;" TEST_PYTHON_PATH)
+    else()
+        list(JOIN TEST_PYTHON_PATH ":" TEST_PYTHON_PATH)
+    endif()
+
     set_tests_properties(${TEST_TARGET}
         PROPERTIES
             ENVIRONMENT
-            "PYTHONPATH=${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/python:${USD_ROOT}/${CMAKE_INSTALL_LIBDIR}/python:$ENV{PYTHONPATH};PXR_PLUGINPATH_NAME=${_TEST_PXR_PLUGIN_PATH}"
```

Comment:
@colevfx  - thanks for fixing up the missing semicolon.  

I was wondering... if there was a cleaner way to express the multiple environment variables we're setting (extending on your great idea around joining token lists).

What would you think of something to this effect:
```cmake
# Create a list of python path values.
set(TEST_PYTHON_PATHS 
    "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/python" 
    "${USD_ROOT}/${CMAKE_INSTALL_LIBDIR}/python" 
    "$ENV{PYTHONPATH}"
)

# Join python path values into a single string.
if (MSVC)
    list(JOIN TEST_PYTHON_PATHS ";" TEST_PYTHON_PATHS)
else()
    list(JOIN TEST_PYTHON_PATHS ":" TEST_PYTHON_PATHS)
endif()

# Join multiple environment variable strings into a single one.
list(JOIN ENV_PATHS ";"
    "PYTHONPATH=${TEST_PYTHON_PATHS}" 
    "PXR_PLUGIN_PATH=${_TEST_PXR_PLUGIN_PATH}"
)

set_tests_properties(${TEST_TARGET}
        PROPERTIES
            ENVIRONMENT
                ${ENV_PATHS}
)
```

This way we can de-couple the _joining of values per-env-var_ from the _joining of different env-vars_, which might clean up the logic a tiny bit.

Reply:
I actually played with this idea myself yesterday!

I decided _not_ to do it then because it seemed to me that if we scaled this pattern to more variables, we'd be adding a set command, a list command, and modifying the final join statement for every variable we are adding. It seemed like an anti-pattern to me to have to do three modifications instead of one.

The advantage to what you wrote is that it's more semantic and easy to read, and truthfully I don't think it's likely we'd be adding more env vars here. But the disadvantage is that if we did adding more env vars would be slightly harder.

If you think that trade off is worth it, I'm happy to revisit this tonight and make this work 😄

Reply:
Thanks @colevfx, we definitely want to strive for more readability here (at the cost of more operations) - so if you could revisit this we'd appreciate it!  This might also fix some of the trailing colons in the test runs.

If we end up with more environment variables, we could try combining initial `set` and `list` into a single utility function `_set_env_var_paths` which should clean that up just a bit further.

Reply:
Gave this a go tonight, but had some serious problems trying to get it working in the manner you originally suggested. So what I tried doing is building out the path more semantically, but keeping the original overall command structure that I had before. I'm hoping that this is in the spirit of what you wanted, making it more explicit what's happening and why, even if it's not in the form you requested exactly. 😄 

The TL;DR on why it wasn't working great was that the when you list(JOIN ...) the multiple environment path strings we were building, it wasn't just gluing the two environment variable paths, but also all of the sub-elements of each list which doesn't work. If need be I can share further details, and below is my non-functioning version of the code that seems okay on Windows, but won't pass Linux tests

>     # Set-up runtime environment variables for the test.
>     # The paths refer to the build tree (which mirrors the final installation).
>     set(TEST_PYTHON_PATHS
>         "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}/python" 
>         "${USD_ROOT}/${CMAKE_INSTALL_LIBDIR}/python" 
>         "$ENV{PYTHONPATH}")
>     
>     if (MSVC)
>         list(JOIN TEST_PYTHON_PATHS ";" TEST_PYTHON_PATHS)
>     else()
>         list(JOIN TEST_PYTHON_PATHS ":" TEST_PYTHON_PATHS)
>     endif()
> 
>     # Join multiple environment variable strings
>     set(ENV_PATHS
>         "PYTHONPATH=${TEST_PYTHON_PATHS}"
>         "PXR_PLUGINPATH_NAME=${_TEST_PXR_PLUGIN_PATH}")
>     list(JOIN ENV_PATHS "\\;" ENV_PATHS) 
> 
>     set_tests_properties(${TEST_TARGET}
>         PROPERTIES
>             ENVIRONMENT
>             "${ENV_PATHS}")

Reply:
Hey @colevfx - hehe that's what I get for proposing untested code.  

Thanks for finding a different and working solution for this, breaking it down into logical blocks, and keeping it readable :-)  It looks great!